src/Disks/ObjectStorages/Local/LocalObjectStorage.cpp

```diff
@@ -56,6 +63,14 @@ std::unique_ptr<ReadBufferFromFileBase> LocalObjectStorage::readObject( /// NOLI
     const ReadSettings & read_settings,
     std::optional<size_t> read_hint) const
 {
+    auto current_context = CurrentThread::getQueryContext();
+    if (current_context && current_context->getSettingsRef()[Setting::check_rights_local_storage].value)
```

Comment:
I do not think this is correct to check this on LocalObjectStorage level, because we can also use it for disks used by MergeTree tables, to which such limitation is not applicable. Why not just put it as suggested in the comment https://github.com/ClickHouse/ClickHouse/pull/86844#discussion_r2333927028?